src/main/groovy/org/shipkit/internal/gradle/plugin/PluginUtil.java

```diff
@@ -0,0 +1,34 @@
+package org.shipkit.internal.gradle.plugin;
+
+import org.gradle.api.Project;
+import org.gradle.api.file.FileTree;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.util.PatternSet;
+import org.shipkit.internal.gradle.util.JavaPluginUtil;
+
+import java.io.File;
+import java.util.Set;
+
+class PluginUtil {
```

Comment:
This class nicely encapsulates plugin complexities, making the code easy to read and maintain. Thanks!